subprojects/core/src/main/java/org/gradle/process/internal/worker/DefaultWorkerProcessBuilder.java

```diff
@@ -100,7 +100,14 @@ public String getBaseName() {
 
     @Override
     public WorkerProcessBuilder applicationClasspath(Iterable<File> files) {
-        GUtil.addToCollection(applicationClasspath, files);
+        for (File file : files) {
+            if (file == null) {
+                throw new IllegalArgumentException("Illegal null value provided in this collection: " + files);
+            }
+            if (file.exists()) {
```

Comment:
If this is also the code path used by things like `JavaExec` and similar, it should also allow wildcards, and then should probably be something along the lines of `file.exists() || file.getName().equals("*")`.

Reply:
Thinks, I will take some spare time rewrite test and the if statement,  What does the `*` wildcards stands for, does it mean for a specific path?

Reply:
`foo/bar/*` on a class path is effectively the same as listing all jars in that directory explicitly. On case-sensitive filesystems like per default on Windows, it should - consider all `jar` and `JAR` if I remember correctly.

Reply:
Sorry misunderstanding  this  as whole path as `*` , Yes, It should be considered.